cfn-resources/teams/cmd/resource/team-user/team_user.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package teamuser
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20231001001/admin"
+)
+
+func FilterOnlyValidUsernames(mongoDBCloudUsersAPIClient TeamUsersAPI, usernames []string) ([]atlasv2.CloudAppUser, *http.Response, error) {
+	var validUsers []atlasv2.CloudAppUser
+	for _, elem := range usernames {
+		userToAdd, httpResp, err := mongoDBCloudUsersAPIClient.GetUserByUsername(context.Background(), elem)
+		if err != nil {
+			_, _ = logger.Warnf("Error while getting the user %s: (%+v) \n", elem, err)
+			return nil, httpResp, err
+		}
+		validUsers = append(validUsers, *userToAdd)
+	}
+	return validUsers, nil, nil
+}
+
+func initUserSet(users []atlasv2.CloudAppUser) map[string]bool {
+	usersSet := make(map[string]bool)
+	for i := 0; i < len(users); i++ {
+		usersSet[*(users[i]).Id] = true
+	}
+	return usersSet
+}
+
+func GetUsersToAddAndRemove(currentUsers []atlasv2.CloudAppUser, newUsers []atlasv2.CloudAppUser) (toAdd []string, toDelete []string, err error) {
```

Comment:
[nit] names that include `And` or `Or` in their name do too many things, consider splitting into two methods that accept both sets

Reply:
I am only changing the name here, agree it wasn't the best. I think it doesn't make sense to split it because it's actually logically connected (it has to calculate users to be removed and users to be added, it goes through the same lists and needs same initialization).

Resolving here, feel free to re-open if you disagree.